lib/services/iotaRedirector.js

```diff
@@ -162,8 +162,15 @@ function splitByProtocol(req) {
             clonedObject.protocol = req.body[collectionName][i].protocol;
             clonedObject.resource = req.protocolObj[clonedObject.protocol].resource;
 
-            collections[req.body[collectionName][i].protocol] = {};
-            collections[req.body[collectionName][i].protocol][collectionName] = [_.clone(req.body[collectionName][i])];
+            if (collections[req.body[collectionName][i].protocol] &&
+                collections[req.body[collectionName][i].protocol][collectionName]) {
+                collections[req.body[collectionName][i].protocol][collectionName].push(
+                    _.clone(req.body[collectionName][i]));
+            } else {
+                collections[req.body[collectionName][i].protocol] = {};
+                collections[req.body[collectionName][i].protocol][collectionName] =
+                    [_.clone(req.body[collectionName][i])];
+            }
```

Comment:
So, if obj+arr already exist, push new item (cloned), else, create obj+array wuth new item (cloned).
Before this change, a second item removed the first, as it seems.
Just trying to understand ...

I imagine `push` works an an empty array also and this could be simplified, something like this:

```
if (!collections[req.body[collectionName][i].protocol] || !collections[req.body[collectionName][i].protocol][collectionName])
  {
    collections[req.body[collectionName][i].protocol] = {};
    collections[req.body[collectionName][i].protocol][collectionName] = [];
  }
  collections[req.body[collectionName][i].protocol][collectionName].push(_.clone(req.body[collectionName][i]));
```

[ Remember, I'm not familiar with node, neither expert in javascript ... ]